jobs/oneTimePopulateTableMapping.js

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/* This script is a point-in-time script that populates any missing URL and URN to blob name mappings in Azure table
+  based on JSON documents stored in Azure blob container.
+
+  The following variables should be set:
+  CRAWLER_STORAGE_ACCOUNT, CRAWLER_STORAGE_NAME, CRAWLER_STORAGE_KEY
+*/
+
+const AzureStorage = require('azure-storage');
+const config = require('painless-config');
+const Q = require('q');
+const qlimit = require('qlimit');
+
+
+const jobConfig = {
+  azureStorage: {
+    account: config.get('CRAWLER_STORAGE_ACCOUNT'),
+    key: config.get('CRAWLER_STORAGE_KEY'),
+    name: config.get('CRAWLER_STORAGE_NAME')
+  }
+};
+
+const stats = {
+  batches: 0,
+  total: 0,
+  url: {
+    added: 0,
+    skipped: 0,
+    skippedReason: {
+      alreadyExists: 0,
+      errors: 0
+    }
+  },
+  urn: {
+    added: 0,
+    skipped: 0,
+    skippedReason: {
+      alreadyExists: 0,
+      errors: 0
+    }
+  }
+};
+const limit = qlimit(100);
+const retryOperations = new AzureStorage.ExponentialRetryPolicyFilter();
+const blobService = AzureStorage.createBlobService(jobConfig.azureStorage.account, jobConfig.azureStorage.key).withFilter(retryOperations);
+const tableService = AzureStorage.createTableService(jobConfig.azureStorage.account, jobConfig.azureStorage.key).withFilter(retryOperations);
+let continuationToken = null;
+
+console.time('Populate table mapping');
+
+start();
+
+function start() {
+  return Q.fcall(() => retrieveBlobNames()
```

Comment:
point of reference, you can also do Q().then(retrieveBlobNames).then... rather than fcall.